harpy/convert.py

```
@@ -0,0 +1,257 @@
+                _id = pysamfq.name.split(":")
+                self.barcode = _id.pop(-1)
+                self.id = ":".join(_id)
+                self.valid = "N" not in self.barcode
+        elif bc in ["haplotagging", "standard"]:
+            # identify the BX:Z SAM tag and remove it from the comment
+            bc = [i for i in self.comment.split() if i.startswith("BX:Z")]
+            if len(bc) < 1:
+                # is invalid
+                self.barcode = INVALID_HAPLOTAGGING
+            else:
+                self.barcode = bc.pop().lstrip("BX:Z:")
```

Ruff (0.8.2)

Using `.strip()` with multi-character strings is misleading

(B005)